

# Draft 1

This pipeline will take in a FASTA file containing the reference genome and a FASTA/FASTQ/BAM file(s) containing the reads. The transcriptome-level BAM files generated by STAR are provided to Salmon for downstream quantification. The results generated by the pipeline will be written to the output directory.

Here is a more detailed description of each step in the pipeline:

1.  Preprocessing: The reads are first decompressed and aligned to the reference genome using STAR.
2. Alignment and quantification: The aligned reads are then quantified using Salmon. Salmon uses a wicked fast gene and isoform quantification relative to the transcriptome.
3.  Quality control: The results of the quantification are then analyzed using RSeQC to assess the quality of the data.
4. Pseudo-alignment and quantification: The results of the quantification can also be used to pseudo-align and quantify the reads using Salmon. This can be useful if you do not have a reference genome available.
4.  Workflow reporting and genomes: The pipeline also generates a report that summarizes the results and includes the reference genome indices/files.

# Draft 2

This pipeline will read the reads from the input FASTQ files, index them with Salmon, quantify them with Salmon, and write the results to a CSV file in the output directory.


# Draft 3
